harness/determined/cli/user.py

```diff
@@ -70,30 +43,22 @@ def log_in_user(parsed_args: Namespace) -> None:
         username = parsed_args.username
 
     message = "Password for user '{}': ".format(username)
-
-    # In order to not send clear-text passwords, we hash the password.
-    password = api.salt_and_hash(getpass.getpass(message))
+    password = getpass.getpass(message)
 
     token_store = authentication.TokenStore(parsed_args.master)
-
     token = authentication.do_login(parsed_args.master, username, password)
     token_store.set_token(username, token)
     token_store.set_active(username)
 
 
 @authentication.optional
+@login_sdk_client
```

Comment:
with this change, when you're already logged out, doing `det user logout` will return:
```
Failed to log out user: {"error":{"code":16,"reason":"Unauthenticated","error":"invalid credentials"}}
```
which I think is not necessary. logout should not "require" user to be logged in.